### PR TITLE
`Development`: Cache windfiles on startup

### PR DESCRIPTION
**Pull request by reschandreas**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).

#### Changes affecting Programming Exercises
- [x] I tested **all** changes and their related features with **all** corresponding user types on Test Server 3 (LocalCI and LocalVC).

### Motivation, Description, and Context

The fetching of the windfiles was pretty flaky on TS3 with multiple nodes, this fixes it by moving the retrieval to the startup of Artemis and throwing errors if the resources could not be found. This PR also removes Kotlin sequential for LocalCI as it is not yet fully supported. It also improves LocalCI in that it does not use root for the execution of the script.

### Steps for Testing

Prerequisites:
- 1 Instructor

1. Log in to Artemis
2. Create Programming Exercises, the build jobs should work as expected

#### Performance Review
- [x] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [x] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

#### Server

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [AeolusTemplateService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5903/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.connectors.aeolus/AeolusTemplateService.html) | 89% | ✅ |
| [LocalCIContainerService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5903/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.connectors.localci/LocalCIContainerService.html) | 70% | ✅ |
| [LocalCIProgrammingLanguageFeatureService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5903/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.connectors.localci/LocalCIProgrammingLanguageFeatureService.html) | 100% | ✅ |

## Summary by CodeRabbit

- **Refactor**
  - Improved logging consistency within the Aeolus Template Service.
  - Enhanced Docker command execution and script creation in Local CI Container Service.
  - Updated programming language feature settings for Kotlin.

- **Chores**
  - Streamlined service initialization with pre-loading scripts into cache.

### CODE REVIEW

**Review by laurenzfb — APPROVED** [review]
demonstrated in testing session, works as expected. Code also LGTM

**Review by mateusmm01 — APPROVED** [review]
Tested on ts3. Works as expected.

**Review by georgyia — APPROVED** [review]
tested manually on ts2, works as expected

**Review by egekurt123 — APPROVED** [review]
manually tested on ts3 and ts5. Works as expected

File: src/main/java/de/tum/in/www1/artemis/service/connectors/aeolus/AeolusTemplateService.java [hunk 1]
```diff
@@ -44,6 +44,35 @@ public class AeolusTemplateService {
...
+                String uniqueKey = directory + "_" + filename;
+                byte[] fileContent = IOUtils.toByteArray(resource.getInputStream());
+                String script = new String(fileContent, StandardCharsets.UTF_8);
+                Windfile windfile = readWindfile(script);
+                this.addInstanceVariablesToWindfile(windfile, ProgrammingLanguage.valueOf(directory.toUpperCase()), optionalProjectType);
+                templateCache.put(uniqueKey, windfile);
+            }
+            catch (IOException | IllegalArgumentException e) {
+                logger.error("Failed to load windfile {}", resource.getFilename(), e);
+            }
```
Comment (bassner) [inline src/main/java/de/tum/in/www1/artemis/service/connectors/aeolus/AeolusTemplateService.java:74]: Question: If the loading of Windfiles was flaky before, is it a good idea to just ignore this error here? What effect would that have on other code that needs these templates later on?

**Review by bassner — APPROVED** [review]
LGTM

**Pull request merged**